neqo-transport/tests/server.rs

```diff
@@ -273,7 +272,7 @@ fn zero_rtt() {
     let t = server.process(None, now).callback();
     now += t;
     assert_eq!(server.process(None, now), Output::None);
-    assert_eq!(server.active_connections().len(), 1);
+    assert_eq!(server.active_connections().len(), 0);
```

Comment:
Previously the `Server` would cache the active connections in `Server::active`. It would only purge this cache on a call to `Server::active_connections`, even though the connection in this case is already closed in `generate_ticket`.

Given that this pull request removes the state tracking (i.e. `Server::active`), `server.active_connections` now correctly returns `0`.